datanode/sqlstore/positions.go

```diff
@@ -70,6 +71,27 @@ func (ps *Positions) GetByMarketAndParty(ctx context.Context,
 		marketID, partyID))
 }
 
+func (ps *Positions) GetByMarketAndParties(ctx context.Context, marketIDRaw string, partyIDsRaw []string) ([]entities.Position, error) {
```

Comment:
If this is going to be called a lot it might be worth figuring out if we can cache somehow in the service, we want as few SQL calls as possible per block as each one adds a few 10s of ms of latency (hence all this batching and caching and flushing)

Reply:
I've added this to the service, using it to first retrieve parties from cache, and add missing ones if needed. This isn't going to be called often, though. The only time this can happen is during MTM, and we will, at most, receive this event once per market, every N seconds (when we MTM)